packages/core/public/internal-base/src/index.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+export type { CoreContext } from './core_context';
+
+// re-export some types from the common package
+export type { PluginName, PluginOpaqueId, CoreId } from '@kbn/core-common-internal-base';
```

Comment:
I'm re-exporting some common package's types from the public/server packages for now. This may change in the future when we will stop re-exporting those from the core endpoints, but for now it will ease the migration by avoiding to import from these two packages at the same time from within core.

Reply:
Is there a chance we avoid the re-export here and just import from the shared common all over the place? Not a blocker for now and maybe something easier to do with eslint rules in place to help \cc @spalger

Reply:
Yea, I'm planning on doing it. Can even be done in this PR before merging if we want. Just wanted to get your team's initial thoughts on the PR first.